src/local-history/local-history-manager.ts

```diff
@@ -27,6 +27,9 @@ export class LocalHistoryManager {
         vscode.workspace.onDidSaveTextDocument((document: vscode.TextDocument) => {
             this.saveEditorContext(document);
         });
+        vscode.workspace.onWillRenameFiles((e: vscode.FileWillRenameEvent) => {
```

Comment:
[question] From vscode API: **onWillRenameFiles "is not fired when files change on disk, e.g triggered by another application"**. Should we switch to fs.watch or DirectoryWatcher (I think this is better than fs.watch) so we can detect if the user renames a file on disk manually?

Reply:
yes that is a good idea, I'll update and use the  watcher api.